File: wooseongshin/2월/10일/타겟넘버.java
class Solution {
    public static int cnt = 0;
    public int solution(int[] numbers, int target) {
        dfs(0,numbers,target,0);

        return cnt;

    }

    public void dfs(int depth,int[] numbers,int target,int result){
        if (depth == numbers.length){
            if(target == result){
                cnt+=1;
            }
            return;
        }
        int add = result + numbers[depth];
        int sub = result - numbers[depth];

        dfs(depth+1,numbers,target,add);
        dfs(depth+1,numbers,target,sub);

    }

}

File: Eunleelee/2024-02/2월 10일/타겟넘버.py
from collections import defaultdict as dd
def solution(numbers, target):
    a = [0]
    d = dd(int)
    d[0] = 1
    for n in numbers:
        nd = dd(int)
        na = []
        for i in a:
            if i+n not in nd:
                na.append(i+n)
            if i-n not in nd:
                na.append(i-n)
            nd[i+n] += d[i]
            nd[i-n] += d[i]
        d = nd
        a = na
    return d[target]